orttraining/orttraining/python/orttraining_pybind_state.cc

```diff
@@ -357,16 +358,95 @@ void addObjectMethodsForTraining(py::module& m, ExecutionProviderRegistrationFn
       .def("reserve", [](std::vector<OrtValue>* v, const size_t len) { v->reserve(len); })
       .def("shrink_to_fit", [](std::vector<OrtValue>* v) { v->shrink_to_fit(); })
       .def("__len__", [](const std::vector<OrtValue>& v) { return v.size(); })
-      .def("__iter__", [](const std::vector<OrtValue>& v) {
-        return py::make_iterator(v.cbegin(), v.cend());
-      },
-           py::keep_alive<0, 1>())
+      .def(
+          "__iter__", [](const std::vector<OrtValue>& v) {
+            return py::make_iterator(v.cbegin(), v.cend());
+          },
+          py::keep_alive<0, 1>())
       .def("__getitem__", [](const std::vector<OrtValue>& v, const size_t idx) {
         return v.at(idx);
       })
       .def("dlpack_at", [](std::vector<OrtValue>* v, const size_t idx) {
         return py::reinterpret_steal<py::object>(ToDlpack(v->at(idx)));
-      });
+      })
+      .def("to_dlpack", [](const std::vector<OrtValue>& v, py::object to_tensor) -> py::list {
```

Comment:
Rename "to_dlpack" to "to_tensors_of_target_framework" or similar? Because the type of result here depends on the "to_tensor" function passed in.

Reply:
I changed the behaviour of the function, if to_tensor is None, then it returns a capsule for all OrtValue in the vector. I think dlpack should be used somewhere in the name as it relies on it.